packages/bpk-component-scrollable-calendar/src/BpkScrollableCalendarGridList.js

```diff
@@ -49,87 +52,121 @@ class BpkScrollableCalendarGridList extends React.Component {
     );
     const months = getMonthsArray(startDate, monthsCount);
 
-    const cache = new CellMeasurerCache({
-      fixedWidth: true,
-      defaultHeight: 276, // most common height (in px) of BpkScrollableCalendarGrid
+    // Here we calculate the height of each calendar grid item in pixels, as the `react-window` API
+    // requires that these are provided so that they can be efficiently rendered.
+    const monthItemHeights = months.map((month) => {
+      const firstDayOffset = (month.getDay() + 7 - props.weekStartsOn) % 7;
+      const monthLength = DateUtils.daysInMonth(
+        month.getYear(),
+        month.getMonth(),
+      );
+      const calendarGridSpaces = firstDayOffset + monthLength;
+      const rowCount = Math.ceil(calendarGridSpaces / COLUMN_COUNT);
+      return BASE_MONTH_ITEM_HEIGHT + ROW_HEIGHT * rowCount;
     });
+
     this.state = {
       months,
-      cache,
+      monthItemHeights,
+      outerHeight: ESTIMATED_MONTH_ITEM_HEIGHT,
     };
   }
```

Comment:
could this be moved to a separate class/function/custom hook to be able to test it on isolation? and add a test to it

Reply:
I think I would also wrap the call to such a function in a useMemo hook, to cache the calculation when possible

Reply:
I will double check on this as I am reverting previous work we have done back to what it once was back a few months ago! 

So will be putting it all back as it was but will see what can be done to improve it